build.cake

```diff
@@ -91,10 +104,12 @@ Task("TestUsingExampleImplementation")
         DotNetCoreBuild(allSln, dotNetCoreBuildSettings);
 
         var allProjects = GetFiles(buildDir + "/*/*.csproj");
+        var parallelOptions = new ParallelOptions 
+        {
+            MaxDegreeOfParallelism = 16
```

Comment:
I wasn't super happy with it, as it seems like a relatively arbitrary number, but without it I got an error from Travis saying he ran out of memory -- presumably because he spun up too many threads and started doing more work than he could handle.

I then wanted to set it equal to Environment.ProcessorCount, but noticed the build slowed down (still faster than what we had, but slower than some other tests I had done with 30.. 20..). You can see this answer for a bit more clarification (https://stackoverflow.com/questions/20806238/maxdegreeofparallelism-environment-processorcount-slows-down-execution-time-on)